jwala-services/src/main/java/com/cerner/jwala/control/webserver/command/WebServerCommandFactory.java

```diff
@@ -71,36 +73,87 @@ public void listCommands() {
+        commands.put(WebServerControlOperation.START.getExternalValue(), webServer -> {
+            final String startScriptName = START_SCRIPT_NAME.getValue();
+            checkExistsAndCopy(webServer, startScriptName);
+            return remoteCommandExecutorService.executeCommand(
+                    new RemoteExecCommand(getConnection(webServer), getShellCommand(startScriptName,
+                            webServer,
+                            webServer.getName())));
+        });
+
+
+        commands.put(WebServerControlOperation.STOP.getExternalValue(), webServer -> {
+            final String stopScriptName = STOP_SCRIPT_NAME.getValue();
+            checkExistsAndCopy(webServer, stopScriptName);
+            return remoteCommandExecutorService.executeCommand(
+                    new RemoteExecCommand(getConnection(webServer), getShellCommand(stopScriptName,
+                            webServer,
+                            webServer.getName())));
+        });
+
+        commands.put(WebServerControlOperation.DELETE_SERVICE.getExternalValue(), webServer -> {
+            final String deleteServiceScriptName = DELETE_SERVICE_SCRIPT_NAME.getValue();
+            checkExistsAndCopy(webServer, deleteServiceScriptName);
+            return remoteCommandExecutorService.executeCommand(
+                    new RemoteExecCommand(getConnection(webServer), getExecCommand(deleteServiceScriptName,
+                            webServer)));
+        });
+
+        commands.put(WebServerControlOperation.INSTALL_SERVICE.getExternalValue(), webServer -> {
+            final String installServiceWsScriptName = INSTALL_WS_SERVICE_SCRIPT_NAME.getValue();
+            checkExistsAndCopy(webServer, installServiceWsScriptName);
+            return remoteCommandExecutorService.executeCommand(
+                    new RemoteExecCommand(getConnection(webServer),
+                            getShellCommand(installServiceWsScriptName,
+                                    webServer,
+                                    ApplicationProperties.getRequired(PropertyKeys.REMOTE_PATHS_APACHE_HTTPD_CONF),
+                                    ApplicationProperties.getRequired(PropertyKeys.REMOTE_PATHS_APACHE_HTTPD))));
+        });
+
         commands.put(WebServerControlOperation.VIEW_HTTP_CONFIG_FILE.getExternalValue(), (WebServer webServer)
                 -> remoteCommandExecutorService.executeCommand(
                 new RemoteExecCommand(getConnection(webServer),
-                        new ExecCommand("cat", ApplicationProperties.getRequired(PropertyKeys.REMOTE_PATHS_APACHE_HTTPD_CONF)+"/httpd.conf"))));
+                        new ExecCommand("cat", ApplicationProperties.getRequired(PropertyKeys.REMOTE_PATHS_APACHE_HTTPD_CONF) + "/httpd.conf"))));
+
+    }
+
+    private void checkExistsAndCopy(WebServer webserver, String scriptName) {
```

Comment:
I was wondering since I heard Arvindo say something about java docs the other day, do we put them in every method we create ? Or it depends whether it's private or not ? Or depends on the complexity of the method ?